### ISSUE TITLE
Merge with reference fails if entity is associated to a column (non id)

### ISSUE THREAD

**Issue by DavideD** [hibernate/hibernate-reactive#1504]
See the following tests:
* https://github.com/hibernate/hibernate-reactive/blob/main/hibernate-reactive-core/src/test/java/org/hibernate/reactive/LazyUniqueKeyTest.java#L71
* https://github.com/hibernate/hibernate-reactive/blob/main/hibernate-reactive-core/src/test/java/org/hibernate/reactive/EagerUniqueKeyTest.java#L71

I've created the following testcase for ORM: https://github.com/hibernate/hibernate-orm/pull/6239
Waiting for ORM to fix https://hibernate.atlassian.net/browse/HHH-16298

(Follows https://github.com/hibernate/hibernate-reactive/pull/1498)

**Comment by blafond** [comment]
@DavideD I changed the EntityTypes.getIdentifier() method to mimic your changes in ORM's [EntityType](https://github.com/hibernate/hibernate-orm/pull/6495/files). I also tweaked our failing merge tests to mimic the new ORM merge test.

lates:  https://github.com/blafond/hibernate-reactive/commits/1504-merge-reference

Now all Eager & Lazy merge tests fail because the `initializer.getImplementation()` forces a call to the reactive session's `immediateLoad()`.

I'm assuming I need to route it to our `reactiveImmediateLoad()` to make this work, but right now it's only called in `reactiveFetch()`?
